engine/class_modules/priest/sc_priest.hpp

```diff
@@ -983,30 +982,48 @@ struct fiend_melee_t : public priest_pet_melee_t
   }
 };
 
+// ==========================================================================
+// Shadowflame Rift
+// ==========================================================================
+struct shadowflame_rift_t final : public priest_pet_spell_t
+{
+  shadowflame_rift_t( base_fiend_pet_t& p ) : priest_pet_spell_t( "shadowflame_rift", &p, p.o().find_spell( 344748 ) )
+  {
+    background = true;
+  }
+};
+
 // ==========================================================================
 // Shadowflame Prism
 // ==========================================================================
 struct shadowflame_prism_t final : public priest_pet_spell_t
 {
-  double shadowflame_increase;
+  double duration;
```

Comment:
please use timespan_t here. That way the units are handled at the point of initialization and not during usage of the value.